Source/stochastic_internal.cpp

```diff
@@ -25,7 +25,7 @@ bool stochastic_internal::evaluate_probability(double probability) {
     return R < probability;
 }
 
-size_t stochastic_internal::weight_proportionate_selection(double* weights, size_t num_weights) {
+size_t stochastic_internal::weight_proportionate_selection(const double* const weights, size_t num_weights) {
```

Comment:
Should this method be marked `const`?

Reply:
This one probably could be marked const. In general the stochastic methods can't be made const because the advance the mersenne twister to the next number in the sequence.